connectors/sources/servicenow.py

```diff
@@ -194,6 +217,83 @@ async def filter_services(self):
             logger.exception(f"Error while filtering services. Exception: {exception}.")
             raise
 
+    async def fetch_data(self, url, params):
+        """Hit API call and iterate over the fetched response.
+
+        Args:
+            url (str): ServiceNow query to be executed.
+            params (dict): Params to format the query.
+
+        Yields:
+            dict: Formatted document.
+        """
+
+        try:
+            async for response in self._api_call(url=url, params=params):
+                for record in response:  # pyright: ignore
+                    yield record
+
+        except Exception as exception:
+            logger.warning(
+                f"Skipping data for {url} with {params}. Exception: {exception}."
+            )
```

Comment:
question: Should we either re-raise the exception so the data source can also handle it (and maybe set an error?) or simply let the data source handle it by moving the log message?

Reply:
Moving exception handling to data source and as user must be warned for that skipped data kept `logger.warning` rather then `logger.error`. Should I replace to error?